lib/chai/core/assertions.js

```diff
@@ -215,6 +215,16 @@ module.exports = function (chai, _) {
   Assertion.addChainableMethod('contains', include, includeChainingBehavior);
   Assertion.addChainableMethod('includes', include, includeChainingBehavior);
 
+  // TODO: Add jsdoc
+
+  function assertFail (msg) {
+    if (msg) flag(this, 'message', msg);
+    var obj = flag(this, 'object');
+    throw new chai.AssertionError(msg, {}, obj);
+  }
+
+  Assertion.addMethod('fail', assertFail);
+
```

Comment:
Shouldn't matter that `expect` is a function. Fail can just be added as a property of it. See the example below:

```js
> function x() {}
undefined
> x.fail = function () {}
[Function]
> x
{ [Function: x] fail: [Function] }
> x.fail
[Function]
>
```